certbot/tests/main_test.py

```diff
@@ -1459,6 +1459,24 @@ def test_plugin_selection_error(self, mock_run, mock_choose):
         # without installer
         self.assertFalse(mock_run.called)
 
+    @mock.patch('certbot._internal.main.updater.run_renewal_deployer')
+    @mock.patch('certbot._internal.plugins.selection.choose_configurator_plugins')
+    @mock.patch('certbot._internal.main._init_le_client')
+    @mock.patch('certbot._internal.main._get_and_save_cert')
+    def test_renew_doesnt_restart_on_dryrun(self, mock_get_cert, mock_init, mock_choose,
+                                            mock_run_renewal_deployer):
+        """A dry-run renewal shouldn't try to restart the installer"""
+        self.config.dry_run = True
+        installer = mock.MagicMock()
+        mock_choose.return_value = (installer, mock.MagicMock())
```

Comment:
https://github.com/certbot/certbot/blob/3d0dad8718e3b1904c1acb7da530a08053ced08f/certbot/certbot/_internal/plugins/selection.py#L185

These should be the other way around